tg_bot/modules/muting.py

```diff
@@ -54,6 +55,105 @@ def mute(bot: Bot, update: Update, args: List[str]) -> str:
+            message.reply_text("I can't seem to find this user")
+            return ""
+        else:
+            raise
+
+    if is_user_admin(chat, user_id, member):
+        message.reply_text("Afraid I can't stop an admin from talking!")
+        return ""
+
+    if user_id == bot.id:
+        update.effective_message.reply_text("I'm not muting myself!")
+        return ""
+
+    split_reason = reason.split(None, 1)
+    if not reason:
+        message.reply_text("You haven't specified a time to mute this user for!")
+        return ""
+
+    else:
+        time_val = split_reason[0].lower()
+        if len(split_reason) > 1:
+            reason = split_reason[1]
+        else:
+            reason = ""
+
+    if any(time_val.endswith(unit) for unit in ('m', 'h', 'd')):
+        unit = time_val[-1]
+        time_num = time_val[:-1]
+        if not time_num.isdigit():
+            message.reply_text("Invalid time amount specified.")
+            return ""
+
+        if unit == 'm':
+            mutetime = int(time.time() + int(time_num) * 60)
+        elif unit == 'h':
+            mutetime = int(time.time() + int(time_num) * 60 * 60)
+        elif unit == 'd':
+            mutetime = int(time.time() + int(time_num) * 24 * 60 * 60)
+        else:
+            # if it gets to this stage, goodluck debugging cause I definitely can't
+            return ""
+
+    else:
+        message.reply_text("Invalid time type specified. Expected m,h, or d, got: {}".format(time_val[-1]))
+        return ""
+
+    log = "<b>{}:</b>" \
+          "\n#TEMP MUTED" \
+          "\n<b>Admin:</b> {}" \
+          "\n<b>User:</b> {}" \
+          "\n<b>Time:</b> {}".format(html.escape(chat.title), mention_html(user.id, user.first_name),
+                                     mention_html(member.user.id, member.user.first_name), time_val)
+    if reason:
+        log += "\n<b>Reason:</b> {}".format(reason)
+
+    try:
+        if member:
+            if member.can_send_messages is None or member.can_send_messages:
+                bot.restrict_chat_member(chat.id, user_id, until_date=mutetime, can_send_messages=False)
+                message.reply_text("Muted! User will be muted for {}.".format(time_val))
+                return log
+
+            else:
+                message.reply_text("This user is already muted!")
+        else:
+            message.reply_text("This user isn't in the chat!")
+
+    except BadRequest as excp:
+        if excp.message == "Reply message not found":
+            message.reply_text("Muted! User will be muted for {}.".format(time_val), quote=False)
+            return log
+        else:
+            LOGGER.warning(update)
+            LOGGER.exception("ERROR muting user %s in chat %s (%s) due to %s", user_id, chat.title, chat.id,
+                             excp.message)
+            message.reply_text("Guess the user can keep babbling since I wasn't able to mute them")
+
+    return ""
+
```

Comment:
This entire function is duplicate code galore, copy pasting my tban and changing the api call just means theres now two things that need to be changed in the event of a bug

Reply:
Basically, extract the logic to a helper func that both tban and tmute use